docs/how-tos/python_consume.md

```diff
@@ -0,0 +1,139 @@
+# Consume GraphQL API using Python
+
+## Preface 
+
+This guide will explain how to fetch data from a GraphQL endpoint and map the response to a 
+Pydantic model. 
+We will cover: 
+
+- Defining a Pydantic model in [Define the Pydantic Model](#define-the-pydantic-model)
+- Fetching the data from GraphQL API in [Fetch Data from GraphQL API](#fetch-data-from-graphql-api)
+- Mapping the response to the Pydantic model in [Map Response Pydantic Model](#map-response-to-pydantic-model)
+
+## Dependencies 
+
+This guide will utilize the following dependencies:
+
+- `requests` to send HTTP request to the GraphQL endpoint
+- `pydantic` for validating and parsing the fetched data
+
+## Define the Pydantic Model
+
+Pydantic is a data validation and parsing library in Python. It provides a efficient way to 
+validate and parse data into Python objects. We can use pydantic models to define the structure and type
+of data we intend to fetch from the enpoint. It automatically validates the input data based on 
+the defined type annotations. If the data does not match the expected types, Pydantic raises errors 
+ensuring the integrity of the data. 
+
+To defind a Pydantic model, we define a class with Pydantic's `BaseModel` class as the base class. 
+We can define a `Person` class with the attributes we intend to fetch from the endpoint. 
+
+!!! example "Define Pydantic Model"
+
+    ```python
+    from pydantic import BaseModel
+
+    class Person(BaseModel):
+        id: int
+        first_name: str
+        last_name: str    
+
+    ```
+!!! info
+
+    Refer to the [How to Guide for Python Service docs](https://diamondlightsource.github.io/graph-federation/how-tos/python_service/) to learn more about the GraphQL API structure
```

Comment:
Think we can probably replace this with a more generic link to the structure of GraphQL APIs